### PR TITLE
raftstore-v2: make coprocessor work

### PR DESCRIPTION
**Pull request by BusyJay**
### What is changed and how it works?
Issue Number: Ref #12842 

What's Changed:

```commit-message
Coprocessor is necessary for a working daemon. This PR adjusts
coprocessor to make it work with raftstore v2. And split check is
also added for auto splitting.
```

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by BusyJay** [comment]
Turns out #13901 was just merged to master (I missed that PR :( ). This PR can reuse most of the observers.

**Review by tabokie — APPROVED** [review]

File: components/raftstore/src/store/worker/split_check.rs [hunk 1]
```diff
@@ -219,32 +220,40 @@ impl Display for Task {
...
 
-pub struct Runner<E, S>
-where
-    E: KvEngine,
-{
-    engine: E,
+pub struct Runner<EK: KvEngine, S> {
+    // We can't just use `TabletRegistry` here, otherwise v1 may create many
+    // invalid records and cause other problems.
+    engine: Either<EK, TabletRegistry<EK>>,
```
Comment (Connor1996) [inline components/raftstore/src/store/worker/split_check.rs:226]: why term `Tablet` is introduced in raftstore v1?

Comment (BusyJay) [reply components/raftstore/src/store/worker/split_check.rs:226]: V2 reuses the worker to do automatically split.

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**